Lodash isEqual. actionDefinition. Publishing fix

rkuffer opened the pull request:
Tests fixes

brandonsechet commented on bot/admin/web/package.json:
```diff
@@ -48,6 +48,7 @@
     "ionicons": "2.0.1",
     "jsoneditor": "^9.5.6",
     "leaflet": "1.2.0",
+    "lodash-es": "^4.17.21",
```
Je me demande si c'est pertinant d'ajouter lodash juste pour isEqual. ON pourrait utiliser structuredClone du navigateur directement. Au final il faut juste mettre à jour @types/node à niveau pour en profiter et je ne crois pas que ça ait d'impact sur notre version d'angular

rkuffer replied:
ben changer la compatibilité cible de l'appli est une décision qui devrait se discuter avec la communauté... Quand à Lodash il est déjà largement utilisé comme dépendance secondaire. De plus  j'utilise la version ES de Lodash. Seule la méthode isEqual et les quelques méthodes qu'elle emploi seront conservées par webpack après tree shaking. L'impact est mineur sur le la taille des packages finaux. Il pourra être remplacé à terme si la target évolue mais en l'état cela permet de ne pas être trop intrusif

rkuffer replied:
Si tu veux avancer sur ce point propose un ticket sur Tock upstream

rkuffer replied:
j'ai pas fait gaffe mais structuredClone ne répond pas du tout au besoin. Ca sert à deep cloner un objet, pas à comparer deux objets

brandonsechet commented:
C'est normal d'avoir importé le package-lock ou c'est un oublie ?

brandonsechet commented on bot/admin/web/src/app/scenarios/scenario-designer/scenario-conception/scenario-conception-item.component.spec.ts:
```diff
@@ -170,28 +170,6 @@ xdescribe('ScenarioConceptionItemComponent', () => {
     expect(component.draggable.data).toEqual(2);
   });
 
-  // it('Should not collect existing utterances on init if item is not an intent', () => {
```
J'avais commenté les tests car l'objet n'était pas correcte et que l'on ne pouvait plus les lancer sans corrections

rkuffer replied:
Une récente refacto rend caduque ce test. Je l'ai supprimé

brandonsechet commented on bot/admin/web/src/app/scenarios/scenario-designer/scenario-conception/scenario-conception-item.component.spec.ts:
```diff
@@ -288,21 +266,6 @@ xdescribe('ScenarioConceptionItemComponent', () => {
   });
 
-  describe('setItemIntentDefinition', () => {
-    // it('Should collect intent utterances after beeing called', () => {
```
Même remarque qu'au dessus

rkuffer replied:
Idem que ci-dessus

rkuffer commented:
C'est normal dans la mesure ou une dépendance a été ajoutée

Le lun. 24 oct. 2022 à 14:03, brandonsechet ***@***.***> a
écrit :

> C'est normal d'avoir importé le package-lock ou c'est un oublie ?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/CreditMutuelArkea/tock/pull/118#issuecomment-1288929097>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYIVMCPZSN45H4EADC3IFZTWEZ3DTANCNFSM6AAAAAARM26C54>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 
*Ce message et toutes les pièces jointes (ci-après le "message") sont 
confidentiels et établis à l'intention exclusive de ses destinataires. 
Toute utilisation ou diffusion non autorisée est interdite. Tout message 
étant susceptible d'altération, l'émetteur décline toute responsabilité au 
titre de ce message s'il a été altéré, déformé ou falsifié. **__*This 
message and any attachments (the "message") are confidential and intended 
solely for the addressees. Any unauthorised use or dissemination is 
prohibited. As e-mails are susceptible to alteration, the issuer shall not 
be liable for the message if altered, changed or falsified.

brandonsechet commented:
> C'est normal dans la mesure ou une dépendance a été ajoutée Le lun. 24 oct. 2022 à 14:03, brandonsechet ***@***.***> a écrit :
> […](#)
> C'est normal d'avoir importé le package-lock ou c'est un oublie ? — Reply to this email directly, view it on GitHub <[#118 (comment)](https://github.com/CreditMutuelArkea/tock/pull/118#issuecomment-1288929097)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AYIVMCPZSN45H4EADC3IFZTWEZ3DTANCNFSM6AAAAAARM26C54> . You are receiving this because you authored the thread.Message ID: ***@***.***>
> -- *Ce message et toutes les pièces jointes (ci-après le "message") sont  confidentiels et établis à l'intention exclusive de ses destinataires. Toute utilisation ou diffusion non autorisée est interdite. Tout message  étant susceptible d'altération, l'émetteur décline toute responsabilité au titre de ce message s'il a été altéré, déformé ou falsifié. **__*This  message and any attachments (the "message") are confidential and intended solely for the addressees. Any unauthorised use or dissemination is prohibited. As e-mails are susceptible to alteration, the issuer shall not be liable for the message if altered, changed or falsified.

Pourtant ça fait un moment que l'on supprime ce fichier même en ajoutant de nouvelle dépendance

rkuffer commented:
Oui mais c'est pas une bonne chose. Il faudrait instituer une regle au
niveau de la communauté voulant que le lock n'est commité que s'il y a
ajout ou update. SI ca n'est pas le cas les utilisateurs devraient faire un
npm ci ou discarder leur lock avant de commiter

Le lun. 24 oct. 2022 à 14:11, brandonsechet ***@***.***> a
écrit :

> C'est normal dans la mesure ou une dépendance a été ajoutée Le lun. 24
> oct. 2022 à 14:03, brandonsechet *@*.
>
> *> a écrit : … <#m_-4014968669190460182_> C'est normal d'avoir importé le
> package-lock ou c'est un oublie ? — Reply to this email directly, view it
> on GitHub <#118 (comment)
> <https://github.com/CreditMutuelArkea/tock/pull/118#issuecomment-1288929097>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/AYIVMCPZSN45H4EADC3IFZTWEZ3DTANCNFSM6AAAAAARM26C54
> <https://github.com/notifications/unsubscribe-auth/AYIVMCPZSN45H4EADC3IFZTWEZ3DTANCNFSM6AAAAAARM26C54>
> . You are receiving this because you authored the thread.Message ID: @.*>
> -- *Ce message et toutes les pièces jointes (ci-après le "message") sont
> confidentiels et établis à l'intention exclusive de ses destinataires.
> Toute utilisation ou diffusion non autorisée est interdite. Tout message
> étant susceptible d'altération, l'émetteur décline toute responsabilité au
> titre de ce message s'il a été altéré, déformé ou falsifié. **__*This
> message and any attachments (the "message") are confidential and intended
> solely for the addressees. Any unauthorised use or dissemination is
> prohibited. As e-mails are susceptible to alteration, the issuer shall not
> be liable for the message if altered, changed or falsified.
>
> Pourtant ça fait un moment que l'on supprime ce fichier même en ajoutant
> de nouvelle dépendance
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/CreditMutuelArkea/tock/pull/118#issuecomment-1288940098>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYIVMCKRAE3M37OXLTBSX4LWEZ4FVANCNFSM6AAAAAARM26C54>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 
*Ce message et toutes les pièces jointes (ci-après le "message") sont 
confidentiels et établis à l'intention exclusive de ses destinataires. 
Toute utilisation ou diffusion non autorisée est interdite. Tout message 
étant susceptible d'altération, l'émetteur décline toute responsabilité au 
titre de ce message s'il a été altéré, déformé ou falsifié. **__*This 
message and any attachments (the "message") are confidential and intended 
solely for the addressees. Any unauthorised use or dissemination is 
prohibited. As e-mails are susceptible to alteration, the issuer shall not 
be liable for the message if altered, changed or falsified.
